src/main/java/com/bean00/MessageController.java

```diff
@@ -1,41 +1,78 @@
 package com.bean00;
 
-import java.io.BufferedReader;
 import java.io.IOException;
-import java.io.PrintWriter;
+import java.io.Writer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 
 public class MessageController {
-    private RequestParser parser;
-    private PrintWriter out;
-    private RequestInterpreter interpreter = new RequestInterpreter();
 
-    public MessageController(BufferedReader in, PrintWriter out) {
-        parser = new RequestParser(in);
+    public Response processRequest(Request request) throws IOException {
+        // get the full path
+        String requestURL = request.getRequestURL();
+        Path path = getFullPath(requestURL);
+
+        // set helpful booleans
+        boolean rootDirectoryWasRequested = requestURL.equals("/");
+        boolean fileCanBeFound = checkIfFileCanBeFound(path);
+
+        // get the contents of the file (as a byte array)
+        byte[] body = getBodyFromFile(path, fileCanBeFound);
 
-        this.out = out;
+        // create the appropriate response
+        Response response = buildResponse(rootDirectoryWasRequested, fileCanBeFound, body);
+
+        return response;
     }
 
-    public Request getRequest() throws IOException {
-        Request request = parser.parseRequest();
+    public void writeResponse(Response response, Writer out) throws IOException {
+        String responseString = response.toString();
 
-        return request;
+        out.write(responseString);
+        out.flush();
     }
 
-    public Response interpretRequest(Request request) {
-        String requestURL = request.getRequestURL();
+    private Path getFullPath(String requestURL) {
+        String pathToPublic = "/Users/jonchin/8th-Light/projects/java-server/cob_spec/public";
```

Comment:
I'm assuming this will get replaced in the future with something dynamic, right?

Reply:
Yep